features/api-layer/src/main/java/org/opennms/features/apilayer/utils/ModelMappers.java

```diff
@@ -132,4 +138,45 @@ public static AlarmFeedback toFeedback(org.opennms.features.situationfeedback.ap
                 .withUser(feedback.getUser())
                 .build();
     }
+
+    public static TopologyEdge toEdge(String protocol, OnmsTopologyEdge edge) {
+        ImmutableTopologyEdge.Builder topologyEdge = ImmutableTopologyEdge.newBuilder()
+                .setId(edge.getId())
+                .setProtocol(protocol)
+                .setTooltipText(edge.getToolTipText())
+                .setSource(ImmutableTopologyPort.newBuilder()
+                        .setId(edge.getSource().getId())
+                        .setTooltipText(edge.getSource().getToolTipText())
+                        .setIfIndex(edge.getSource().getIfindex())
```

Comment:
This throws an NPE if the source ifIndex is null - the ImmutableTopologyEdge doesn't require an ifIndex though.

(Similar problem with the target)

Reply:
Oops, originally I had the ifIndex as a mandatory parameter for ports which is why I had the non-null check. I should fix this in the API project to allow that field to be set with a null value.

Reply:
Updated here:
https://github.com/OpenNMS/opennms-integration-api/pull/11